workflow/rules/trimming.smk

```diff
@@ -68,18 +68,18 @@ rule pipe_ptrimmer_se:
 
 rule ptrimmer_se:
     input:
-        "results/trimmed/adapters/{sample}/{unit}.single_R1.fastq.gz"
+        "results/trimmed/adapters/{sample}/{unit}.single_R1.fastq.gz",
+        primers="results/primers/primers.txt",
     output:
         "results/trimmed/primers/{sample}/{unit}.single.fastq.gz",
     params:
-        primers=config["primers"]["trimming"]["primers"],
         output_dir=lambda wc, output: os.path.dirname(output)
     log:
         "logs/ptrimmer/{sample}-{unit}.log"
     conda:
         "../envs/ptrimmer.yaml"
     shell:
-        "ptrimmer -s single -f {input} -a {params.primers} -o {params.output_dir} &> {log} && "
+        "ptrimmer -s single -f {input} -a {input.primers} -o {params.output_dir} &> {log} && "
```

Comment:
`{input}}` will give you the entire input (including primers). Instead, give the fastq input also a name.